packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceStatusText.tsx

```diff
@@ -1,43 +1,59 @@
-import { TrezorDevice } from '@suite-common/suite-types';
 import * as deviceUtils from '@suite-common/suite-utils';
 import { TOOLTIP_DELAY_LONG, TruncateWithTooltip } from '@trezor/components';
 import React, { MouseEventHandler } from 'react';
 import { Translation } from 'src/components/suite';
 import { DeviceConnectionText } from './DeviceConnectionText';
+import { selectLabelingDataForWallet } from 'src/reducers/suite/metadataReducer';
+import { useSelector } from 'src/hooks/suite';
+import { selectDevice } from '@suite-common/wallet-core';
+import { useWalletLabeling } from 'src/components/suite/labeling/WalletLabeling';
 
 type DeviceStatusTextProps = {
-    device: TrezorDevice;
     onRefreshClick?: MouseEventHandler;
-    walletLabel?: string;
 };
 
 type DeviceStatusVisible = {
     connected: boolean;
-    walletLabel?: string;
-    device: TrezorDevice;
 };
 
-const DeviceStatusVisible = ({ connected, walletLabel }: DeviceStatusVisible) => (
-    <DeviceConnectionText
-        variant={connected ? 'primary' : 'tertiary'}
-        icon={connected ? 'LINK' : 'UNLINK'}
-        data-test={connected ? '@deviceStatus-connected' : '@deviceStatus-disconnected'}
-    >
-        {walletLabel ? (
-            <TruncateWithTooltip delayShow={TOOLTIP_DELAY_LONG}>{walletLabel}</TruncateWithTooltip>
-        ) : (
-            <Translation id={connected ? 'TR_CONNECTED' : 'TR_DISCONNECTED'} />
-        )}
-    </DeviceConnectionText>
-);
-
-export const DeviceStatusText = ({
-    device,
-    onRefreshClick,
-    walletLabel,
-}: DeviceStatusTextProps) => {
-    const { connected } = device;
-    const deviceStatus = deviceUtils.getStatus(device);
+const DeviceStatusVisible = ({ connected }: DeviceStatusVisible) => {
```

Comment:
We use walletLabel only here and it doesn't make sense to pass it through multiple components. At the same time, it's even harder to read.